cityofaustin/joplin
# Api page type and description

This changes our  allPages API to include page types and descriptions for page indexes. 

Example graphql query used to build search  indexes.

- I modified the basepage a bit to allow for `page types` and `short descriptions` and filtered out non-`live` pages. 

```
{
  allPages(live: true) {
    edges {
      node {
        id
        title
        janisUrls
        pageType
        description
      }
    }
  }
}
```

To Test: 
- See API queries include all current page data deployed branch's  in Joplin.

Deployed: http://joplin-pr-api-page-type-description.herokuapp.com

URL query:  http://joplin-pr-api-page-type-description.herokuapp.com/api/graphql/?query={%20allPages(live:%20true)%20{%20edges%20{%20node%20{%20id%20title%20janisUrls%20pageType%20description%20}%20}%20}%20}

Graphiql: http://joplin-pr-api-page-type-description.herokuapp.com/api/graphiql/?query=%7B%20allPages(live:%20true)%20%7B%20edges%20%7B%20node%20%7B%20id%20title%20janisUrls%20pageType%20description%20%7D%20%7D%20%7D%20%7D#query=%7B%0A%20%20allPages(live%3A%20true)%20%7B%0A%20%20%20%20edges%20%7B%0A%20%20%20%20%20%20node%20%7B%0A%20%20%20%20%20%20%20%20id%0A%20%20%20%20%20%20%20%20title%0A%20%20%20%20%20%20%20%20janisUrls%0A%20%20%20%20%20%20%20%20pageType%0A%20%20%20%20%20%20%20%20description%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A
